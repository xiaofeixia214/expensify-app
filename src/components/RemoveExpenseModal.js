import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
	<Modal
		isOpen={props.showRemoveExpenseModal}
		contentLabel="Confirmation"
		closeTimeoutMS={100}
		className="modal"
	>
		<p className="modal__body">Do you want to delete this expense?</p>
		<div className="modal__button-bar">
			<button className="button" onClick={props.handleRemoveExpense}>Yes</button>
			<button className="button" onClick={props.hanldeCancelRemoveExpense}>No</button>
		</div>
	</Modal>
);

export default RemoveExpenseModal;