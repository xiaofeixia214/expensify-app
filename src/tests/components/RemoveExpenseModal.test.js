import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';

let handleRemoveExpense, hanldeCancelRemoveExpense, wrapper;

beforeEach(() => {
	handleRemoveExpense = jest.fn();
	hanldeCancelRemoveExpense = jest.fn();
	wrapper = shallow(
		<RemoveExpenseModal 
			handleRemoveExpense={handleRemoveExpense}
			hanldeCancelRemoveExpense={hanldeCancelRemoveExpense}
		/>
	);
});

test('should render RemoveExpenseModal correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should call handleRemoveExpense when click yes button', () => {
	wrapper.find('button').first().simulate('click');
	expect(handleRemoveExpense).toHaveBeenCalled();
});

test('should call hanldeCancelRemoveExpense when click no button', () => {
	wrapper.find('button').at(1).simulate('click');
	expect(hanldeCancelRemoveExpense).toHaveBeenCalled();
});