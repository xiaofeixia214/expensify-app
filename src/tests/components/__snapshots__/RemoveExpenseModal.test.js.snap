// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={100}
  contentLabel="Confirmation"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <p
    className="modal__body"
  >
    Do you want to delete this expense?
  </p>
  <div
    className="modal__button-bar"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
